_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,176 @@
+
+What you see today when you open the Artsy iOS app is a mix of Objective-C, Swift and React Native. Objective-C and Swift continue to provide a lot of over-arching cross-View Controller code. While individual representations of Artsy resources tend to be built in React Native. All of our React Native code uses Relay to handle API integration.
+
+Our core API serves the public facets of our product, many of our own internal applications, and even [some of your own projects](https://developers.artsy.net/). It's built with [Ruby](https://www.ruby-lang.org/en/), [Rack](http://rack.github.io/), [Rails][rails], and [Grape](https://github.com/intridea/grape) serving primarily JSON. The API is hosted on [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Memcached](http://memcached.org) for caching and [Redis](https://redis.io/) for background queues. We used to employ [Apache Solr](http://lucene.apache.org/solr) and even [Google Custom Search](https://www.google.com/cse) for the many search functions, but have since consolidated on [Elasticsearch](https://www.elastic.co).
+
+Most modern code for both the website, and the iOS app use an orchestration layer which is powered by [GraphQL][graphQL]. Our GraphQL server is an [Express](http://expressjs.com) app, using [express-graphql][express-graphql] to provide a single API end-point. The GraphQL API does not access our data directly, but forwards requests to the core API or other services. We have been migrating shared display logic into the GraphQL server, to make it easier to build consistent clients.
+
+We continue to have a [public HAL+JSON API](https://developers.artsy.net) for external developers. This API is in active use for a few production services inside Artsy.
+
+### Partner Facing
+
+There is more to this than just the consumer facing side though, we have a home-grown Content Management System (CMS) for gallery and institution partners. This CMS allows our partners to upload Show, Fair, Artist and Artwork metadata to our API. 
+
+This CMS use stable, mature technologies like [Rails][rails], [Bootstrap][bootstrap], [Turbolinks][turbolinks], [CoffeeScript][coffee] and 
+
+We have an in-house image processing project (rails, sidekiq, redis, rmagick, imagemagick) 
+
+Partners are billed via 
+https://github.com/artsy/induction/
+
+and handled via
+https://github.com/artsy/vibrations/
+
+Fairs? https://github.com/artsy/waves
+
+### Communications
+
+Art collectors inquire on an artworks though our internal messaging system. This can be either thought an API call, or through emails. For 
+
+https://github.com/artsy/radiation
```

Comment:
Radiation is the system that manages conversation between different parties. It receives messages via API or Email, finds or creates the conversation based on email's recipients and forwards them to the proper emails/users in that conversation. It's doesn't know anything about the context of the emails/messages which makes it a generic system for any type of conversation. Currently it's used to manage conversations started by an inquiry on an artwork.